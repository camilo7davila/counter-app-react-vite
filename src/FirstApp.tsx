import { FC } from "react";

interface Props {
  title: string;
  subtitle: number;
}

const newMessage = {
  message: 'Camilo',
  title: 'camilo'
};

const getValue = (): string => {
  return 'Esto se devuelve'
}

export const FirstApp: FC<Props> = ({ title, subtitle}) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      {/* <code>{ JSON.stringify(newMessage) }</code> */}
      {/* <h1>{ getValue() }</h1> */}
      <p>{ subtitle }</p>
    </>
  )
}
