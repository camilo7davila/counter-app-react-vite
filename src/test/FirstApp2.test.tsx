import { render, screen } from '@testing-library/react';
import { FirstApp } from '../FirstApp';

describe('Pruebas en <FistApp />', () => {
  const title = 'Hola soy camilo'
  const subTitle = 1234

  test('Debe de hacer match con el snapshot', () => {
    const { container } = render(<FirstApp subtitle={subTitle} title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar el mensaje "Hola soy camilo"', () => {
    render(<FirstApp subtitle={subTitle} title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
    // screen.debug() muestra el body de la screen actualizado
  });

  test('debe de mostrar el titulo de un h1', () => { 
    render(<FirstApp subtitle={subTitle} title={title} />);
    expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title)
  });

  test('debe de mostrar el subtitle Enviado por props', () => { 
    render(<FirstApp subtitle={subTitle} title={title} />);
    expect(screen.getByText(subTitle).innerHTML).toContain(`${subTitle}`)
  });
})