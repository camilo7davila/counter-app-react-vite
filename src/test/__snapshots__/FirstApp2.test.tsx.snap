// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FistApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy camilo
  </h1>
  <p>
    1234
  </p>
</div>
`;
